### PR TITLE
Integrate discount-code API at tickets/discount-codes/create route

### LINKED ISSUE #798: Implement discount-code API to tickets/discount-codes/create
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

### PR DESCRIPTION
**Pull request by harshitagupta30**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
The PR integrates discount-code API at tickets/discount-codes/create route.

Fixes #798

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-816-open-event-frontend-fossasia.surge.sh

File: app/controllers/events/view/tickets/discount-codes/create.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+const { Controller } = Ember;
+
+export default Controller.extend({
+  actions: {
+    saveCode(code) {
+      this.get('store').createRecord('discount-code');
+      code.save()
+        .then(() => {
+          this.notify.success(this.l10n.t('Discount code has been successfully created.'));
```
Comment (sumedh123) [inline app/controllers/events/view/tickets/discount-codes/create.js]: this.get('notify')

File: app/controllers/events/view/tickets/discount-codes/create.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        })
+        .catch(()=> {
+          this.notify.error(this.l10n.t('An unexpected error has occured. Discount code cannot be created.'));
```
Comment (sumedh123) [inline app/controllers/events/view/tickets/discount-codes/create.js]: Same here.

**Comment by niranjan94** [comment]
Deployment link: https://pr-816-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-816-open-event-frontend-fossasia.surge.sh

File: app/routes/events/view/tickets/discount-codes/create.js [hunk 3]
```diff
@@ -8,26 +7,6 @@ export default Route.extend({
...
-          name : 'Early bird',
-          id   : 1
-        },
-        {
-          name : 'Late entry',
-          id   : 2
-        }
-      ]
-    };
+    return this.get('store').findAll('discountCode');
```
Comment (geekyd) [inline app/routes/events/view/tickets/discount-codes/create.js]: @harshitagupta30 We are creating a discount code not listing the codes. Please check how we are doing it for other routes

**Comment by niranjan94** [comment]
Deployment link: https://pr-816-open-event-frontend-fossasia.surge.sh

File: app/routes/events/view/tickets/discount-codes/create.js [hunk 4]
```diff
@@ -8,26 +8,9 @@ export default Route.extend({
...
-        },
-        {
-          name : 'Late entry',
-          id   : 2
-        }
-      ]
-    };
+    return this.get('store').createRecord('discount-code',{
+      event  : this.modelFor('events.view'),
+      usedFor: 'ticket'
```
Comment (geekyd) [inline app/routes/events/view/tickets/discount-codes/create.js]: Fix lint errors here.

**Comment by geekyd** [comment]
@harshitagupta30 Please check https://travis-ci.org/fossasia/open-event-frontend/builds/270795845#L3622.

Also please implement the edit discount code.

**Comment by niranjan94** [comment]
Deployment link: https://pr-816-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@harshitagupta30 Will you be continuing on this?

**Pull request closed without merging**